recipes/asyncpp/all/conanfile.py

```diff
@@ -0,0 +1,104 @@
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+    }
+
+    @property
+    def _min_cppstd(self):
+        return 20
+
+    @property
+    def _compilers_minimum_version(self):
+        return {
+            "gcc": "13",
+            "clang": "14",
+        }
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            self.options.rm_safe("fPIC")
+
+    def layout(self):
+        # src_folder must use the same source folder name the project
+        cmake_layout(self, src_folder="src")
+
+    def validate(self):
+        # validate the minimum cpp standard supported. For C++ projects only
+        if self.settings.compiler.cppstd:
+            check_min_cppstd(self, self._min_cppstd)
+        minimum_version = self._compilers_minimum_version.get(str(self.settings.compiler), False)
+        if minimum_version and Version(self.settings.compiler.version) < minimum_version:
+            raise ConanInvalidConfiguration(
+                f"{self.ref} requires C++{self._min_cppstd}, which your compiler does not support."
+            )
+
+        if self.settings.compiler == "apple-clang":
+            # As per https://en.cppreference.com/w/cpp/compiler_support, apple-clang 14.0.0 partially supports C++20 but not jthread
+            raise ConanInvalidConfiguration(f"{self.ref} does not support apple-clang compiler, as it lacks jthread support.")
+
+    def build_requirements(self):
+        self.tool_requires("cmake/[>=3.25 <4]")
+
+    def source(self):
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
+
+    def generate(self):
+        tc = CMakeToolchain(self)
+        tc.variables["ASYNCPP_BUILD_TESTS"] = "OFF"
+        tc.variables["ASYNCPP_BUILD_BENCHMARKS"] = "OFF"
+        tc.generate()
+        venv = VirtualBuildEnv(self)
+        venv.generate(scope="build")
+
+    def build(self):
+        cmake = CMake(self)
+        cmake.configure()
+        cmake.build()
+
+    def package(self):
+        copy(self, "LICENSE.md", self.source_folder, os.path.join(self.package_folder, "licenses"))
+        cmake = CMake(self)
+        cmake.install()
+
+        rmdir(self, os.path.join(self.package_folder, "lib", "pkgconfig"))
+        rmdir(self, os.path.join(self.package_folder, "lib", "cmake"))
+        rmdir(self, os.path.join(self.package_folder, "share"))
+        rm(self, "*.la", os.path.join(self.package_folder, "lib"))
+        rm(self, "*.pdb", os.path.join(self.package_folder, "lib"))
+        rm(self, "*.pdb", os.path.join(self.package_folder, "bin"))
+
+    def package_info(self):
+        self.cpp_info.libs = ["asyncpp"]
+        self.cpp_info.set_property("cmake_file_name", "asyncpp")
+        self.cpp_info.set_property("cmake_target_name", "asyncpp")
```

Comment:
I think the best would be to stick with conan's conventions, so that the package name is consistent with most other libraries out there.

The examples and benchmarks are not so important, as they are mostly used for developing the library, and can be changed easily without modifying the conan recipe.

Reply:
I'm not really up to date on conan's conventions, so I'll leave the choice up to you :)

Reply:
Oh, I see. In that case I think I'd stick with `asyncpp::asyncpp`, that would make it more future proof if I want to add something like `asyncpp::network`.